src/components/v1/DepositButton.tsx

```diff
@@ -205,7 +216,16 @@ const DepositButton: React.FC<DepositButtonProps> = ({
               <Button
                 mt={4}
                 onClick={() => deposit(signer!, depositAmount, selectedToken)}
-                isDisabled={!depositAmount || parseFloat(depositAmount) > balance}
+                isDisabled={
+                  !signer || 
+                  !depositAmount || 
+                  parseFloat(depositAmount || "0") <= 0 || 
+                  parseFloat(depositAmount || "0") > balance ||
+                  depositStatus.loading
+                }
```

Comment:
Nit-pick: extract this into a constant for easier debugging

```suggestion
 const isButtonDisabled = !signer || 
                  !depositAmount || 
                  parseFloat(depositAmount || "0") <= 0 || 
                  parseFloat(depositAmount || "0") > balance ||
                  depositStatus.loading;
                  
                isDisabled={isButtonDisabled}
```